multiversx/mx-chain-go
# Guarded txs relayed

## Reasoning behind the pull request
- relay txs V1 and V2 did not consider the guardian use-case
- possibility of network split due to checking the state of an account guardian on the interceptor instead of on execution
  
## Proposed changes
- Take care to consider also guardian signature check on the relay v1 and v2
- Move the checking of active guardian (on account) from interceptor to execution

## Testing procedure
- use relayed txs with guarded and unguarded transactions scenarios and verify the results
- run all previous guardian scenarios and verify results

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
